### PR TITLE
feat(web): add Journal trash view to restore or purge deleted entries

### LINKED ISSUE #301: feat(web): add Journal trash view to restore or purge deleted entries
## Goal
Give users a UI to see soft-deleted journal entries and either restore them or delete them permanently. The backend already exposes restore/purge; this issue adds the missing frontend surface.

## Scope
- Frontend `apps/web/components/screens/JournalScreen.tsx`: a "Trash" toggle/section listing soft-deleted entries (date + time + size)
- Each trashed row offers Restore (`POST /journal/{id}/restore`) and Delete permanently (`DELETE /journal/{id}?purge=true`), with confirm on purge
- Restoring returns the entry to the active list; purging removes it for good
- Backend: add `GET /journal/trash` to list trashed entries (the only missing API piece)

## References
- Builds on #296 (PR #300): soft-delete + restore/purge API already implemented
- API: `DELETE /journal/{id}` (soft / `?purge=true`), `POST /journal/{id}/restore`
- Store: `apps/python/src/journal_store.py` (`soft_delete`/`restore`/`purge`)

## Acceptance Criteria
- [ ] `GET /journal/trash` returns soft-deleted entries with id, date, size
- [ ] A Trash view in the UI lists deleted entries
- [ ] Restore moves an entry back to the active list
- [ ] Delete permanently removes an entry from trash (after confirm)
- [ ] Tests cover the trash listing endpoint

### PR DESCRIPTION
**Pull request by KazusaNakagawa**
## Summary
- `journal_store.list_trashed()` lists soft-deleted entries (shares glob helper with `list_files`)
- API: `GET /journal/trash` (declared before `/journal/{id}` so the static path isn't shadowed)
- Frontend: a Trash toggle in the list top bar; trashed rows offer Restore (`POST /restore`) and Delete permanently (`DELETE ?purge=true`, with confirm)

## Test plan
- [x] `GET /journal/trash` returns soft-deleted entries with id/date/size; excluded from active list
- [ ] Restore returns an entry to the active list; purge removes it for good (after confirm)
- [ ] pytest 661 + vitest 163 pass

Closes #301

## Summary by Sourcery

Add support for viewing and managing soft-deleted journal entries via a dedicated trash listing in the API and web UI.

New Features:
- Expose a /api/journal/trash endpoint that lists soft-deleted journal entries with id, date, and size.
- Add a trash toggle to the Journal screen that shows trashed entries and allows restoring or permanently deleting them.

Enhancements:
- Refactor journal entry listing logic into a shared helper used for both active and trashed entries.

Tests:
- Add API tests covering trash listing behavior, empty trash responses, and authentication requirements for accessing trash.

## Summary by CodeRabbit

* **New Features**
  * Added a Trash view for journal entries, letting you browse soft-deleted items separately from active entries.
  * Added actions to restore entries or permanently delete them from the Trash.
* **Bug Fixes**
  * Journal entry lists now load consistently for both active and deleted items, with newest entries shown first.
  * The Trash view correctly handles empty states and requires sign-in before access.

### CODE REVIEW

File: apps/web/components/screens/JournalScreen.tsx [hunk 1]
```diff
@@ -155,6 +171,48 @@ export function JournalScreen() {
...
+      } catch (e) {
+        setEntriesError(`Permanent delete failed: ${String(e)}`)
+      }
+    },
+    [loadTrash],
+  )
+
+  // Toggle the trash view, loading its contents when opening.
+  const toggleTrash = () => {
+    setShowTrash((cur) => {
```
[the remark this answers is not in the document]
Comment (KazusaNakagawa) [reply apps/web/components/screens/JournalScreen.tsx]: @sourcery-ai Refactored `toggleTrash` into a `useCallback` that computes `next` outside the updater, sets state, then conditionally loads the trash — side effects are now out of the state updater. Done in 8109264.

File: apps/web/components/screens/JournalScreen.tsx [hunk 2]
```diff
@@ -136,6 +138,20 @@ export function JournalScreen() {
...
+      if (!res.ok) {
+        setEntriesError(`Failed to load trash (HTTP ${res.status})`)
+        return
+      }
+      const data = (await res.json()) as { entries: JournalEntry[] }
+      setTrash(data.entries)
+    } catch (e) {
+      setEntriesError(`Failed to load trash: ${String(e)}`)
+    }
+  }, [])
```
[the remark this answers is not in the document]
Comment (KazusaNakagawa) [reply apps/web/components/screens/JournalScreen.tsx:154]: @coderabbitai Added `setEntriesError(null)` at the start of `loadTrash` so stale errors no longer persist after a successful load. Done in 8109264.

File: apps/web/components/screens/JournalScreen.tsx [hunk 3]
```diff
@@ -266,19 +324,79 @@ export function JournalScreen() {
...
+                            className="rounded-md border px-2 py-1 text-xs text-muted-foreground transition-colors hover:bg-accent hover:text-destructive"
+                          >
+                            Delete
+                          </button>
+                        </div>
+                      </td>
+                    </tr>
+                  ))}
+              </tbody>
+            </table>
```
[the remark this answers is not in the document]
Comment (KazusaNakagawa) [reply apps/web/components/screens/JournalScreen.tsx:402]: @coderabbitai Added a Size (KB) column to the trash table, rendering `e.size` with the same formatting as the active list. Done in 8109264.

**Comment by KazusaNakagawa** [comment]
@sourcery-ai review — addressed all three review comments in 8109264 (trash size column, loadTrash error reset, toggleTrash refactor).
